features/mbedtls/targets/TARGET_CRYPTOCELL310/aes_alt.c

```diff
@@ -707,20 +715,30 @@ int mbedtls_aes_setkey_sw_enc( struct sw_aes_context *ctx, const unsigned char *
 
     switch( keybits )
     {
-        case 128: ctx->nr = 10; break;
-        case 192: ctx->nr = 12; break;
-        case 256: ctx->nr = 14; break;
+        case 128: ctx->keysize_nr = 10; break;
+        case 192: ctx->keysize_nr = 12; break;
+        case 256: ctx->keysize_nr = 14; break;
         default : return( MBEDTLS_ERR_AES_INVALID_KEY_LENGTH );
     }
 
+    if ( NULL == ctx->buf )
+    {
+        ctx->buf = mbedtls_calloc( sizeof(uint32_t) , 68 );
```

Comment:
Do you really need to allocate 68 `uint32_t`s in here? Since you are doing this dynamically cant you just allocate as many bytes as needed taking into account the `keysize_nr`?